src/libraries/System.Security.Cryptography.X509Certificates/tests/DynamicChainTests.cs

```diff
@@ -649,5 +829,115 @@ private static X509Certificate2 TamperSignature(X509Certificate2 input)
             cert[cert.Length - 1] ^= 0xFF;
             return new X509Certificate2(cert);
         }
+
+        private static X509Extension BuildPolicyConstraints(
+            int? requireExplicitPolicySkipCerts = null,
+            int? inhibitPolicyMappingSkipCerts = null
+        )
+        {
+            // RFC 5280 4.2.1.11
+            //    id-ce-policyConstraints OBJECT IDENTIFIER ::=  { id-ce 36 }
+            //    PolicyConstraints ::= SEQUENCE {
+            //         requireExplicitPolicy           [0] SkipCerts OPTIONAL,
+            //         inhibitPolicyMapping            [1] SkipCerts OPTIONAL }
+            //    SkipCerts ::= INTEGER (0..MAX)
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+            using (writer.PushSequence())
+            {
+                if (requireExplicitPolicySkipCerts.HasValue)
+                {
+                    Asn1Tag tag = new Asn1Tag(TagClass.ContextSpecific, 0);
+                    writer.WriteInteger(requireExplicitPolicySkipCerts.Value, tag);
+                }
+
+                if (inhibitPolicyMappingSkipCerts.HasValue)
+                {
+                    Asn1Tag tag = new Asn1Tag(TagClass.ContextSpecific, 1);
+                    writer.WriteInteger(inhibitPolicyMappingSkipCerts.Value, tag);
+                }
+            }
+
+            // Conforming CAs MUST mark this extension as critical.
+            return new X509Extension("2.5.29.36", writer.Encode(), critical: true);
+        }
+
+        private static X509Extension BuildPolicyByIdentifiers(params string[] policyOids)
+        {
+            // id-ce-certificatePolicies OBJECT IDENTIFIER ::=  { id-ce 32 }
+
+            // anyPolicy OBJECT IDENTIFIER ::= { id-ce-certificatePolicies 0 }
+
+            // CertificatePolicies ::= SEQUENCE SIZE (1..MAX) OF PolicyInformation
+
+            // PolicyInformation ::= SEQUENCE {
+            //      policyIdentifier   CertPolicyId,
+            //      policyQualifiers   SEQUENCE SIZE (1..MAX) OF
```

Comment:
Perhaps a test where the sequence is empty?

Reply:
(Maybe it's not needed, but it's sort of a "there's one test for garbage data, do we need to care if this edge case is garbage on all platforms?")

Reply:
Mayyybe? The test for the malformed data is for the policy constraint extension, not the policy identifier extension. 

I'm also not entirely sure how much "documenting" of OS-specific behavior we want to do. For example, right now macOS does not error if this sequence is empty, assuming there is no other reason for it to error. If a later macOS version changed it to start throwing because they started doing tighter asn.1 validation, would we be able to, or care enough, to make changes to the product code to fix the test? Or we just "fix" the test and say "we defer to the platform"?

Reply:
Yep, makes sense, and probably the position I'd've had if I wrote the code instead of read the code.